File: lifecare_app/lib/data/models/hospital_model.dart
class HospitalModel {
  final String id, name, location, address;
  final num pinCode, phone;

  HospitalModel({
    this.id = "",
    required this.name,
    required this.phone,
    required this.location,
    required this.address,
    required this.pinCode,
  });

  factory HospitalModel.fromMap(Map<String, dynamic> json) => HospitalModel(
        id: json["_id"],
        name: json["name"],
        phone: json["phone"],
        location: json["location"],
        address: json["address"],
        pinCode: json["pinCode"],
      );

  factory HospitalModel.initial() => HospitalModel(
        id: "",
        name: "",
        address: "",
        location: "",
        phone: 0,
        pinCode: 0,
      );

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['_id'] = id;
    data['name'] = name;
    data['address'] = address;
    data['location'] = location;
    data['phone'] = phone;
    data['pinCode'] = pinCode;
    return data;
  }
}

File: lifecare_server/models/hospital_model.js
const mongoose = require("mongoose");

const hospitalSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
  },
  address: {
    type: String,
    required: true,
  },
  location: {
    type: String,
    required: true,
  },
  phone: {
    type: Number,
    required: true,
  },
  pinCode: {
    type: Number,
    required: true,
  },
});

const hospital = mongoose.model("hospital", hospitalSchema);

module.exports = {
  hospital,
};
